### PR TITLE
Simplify sum_b() and sum_d() idct functions

### PR DESCRIPTION
**Pull request by jamrial**
Should help compilers generate better code.

### CODE REVIEW

File: libdcadec/idct_fixed.c [hunk 1]
```diff
@@ -28,11 +28,9 @@ static void sum_a(const int * restrict input, int * restrict output, int len)
 
 static void sum_b(const int * restrict input, int * restrict output, int len)
 {
-    for (int i = 0; i < len; i++) {
-        if (i > 0)
-            output[i] = input[2 * i] + input[2 * i - 1];
-        else
-            output[i] = input[2 * i];
+    output[0] = input[0];
+    for (int i = 1; i < len; i++) {
```
Comment (kasper93) [inline libdcadec/idct_fixed.c]: If I may say something. Braces should be removed if we want to keep the style consistent.

**Comment by foo86** [comment]
Does this make any practical difference with recent GCC? I'd expect the compiler to unroll these loops completely. But the change itself improves readability a bit so it can be applied.

**Comment by jamrial** [comment]
Yeah, GCC 4.9.2 x86_64 didn't unroll the loops before this patch on the couple tests i did.

**Comment by foo86** [comment]
Merged, thanks.

**Pull request closed without merging**